hpctestlib/microbenchmarks/gpu/gpu_burn.py

```diff
@@ -52,9 +52,9 @@ def setup_build(self):
 
         if self.gpu_build is None:
             # Try to set the build type from the partition features
-            if 'cuda' in curr_env.features:
+            if 'cuda' in curr_part.features:
                 self.gpu_build = 'cuda'
-            elif 'hip' in curr_env.features:
+            elif 'hip' in curr_part.features:
```

Comment:
I think here the comment must be updated instead.

Reply:
I think it mostly depends on how you want to define it in settings.py, 

from settings.py (here it's an AMD GPU)
```
'partitions': [
                {
                    'name': 'gpu',
                    'descr':'GPU nodes',
                    'scheduler': 'squeue',
                    'launcher': 'srun',
                    'access': [],
                    'environs': ['PrgEnv-cray'],
                    'max_jobs':100,
                    'access': ['-C MI250','-t 00:10:00'],
                    'features' : ['hip'],
```

HIP can be used to compile code that will run on an AMD GPU or nvidia GPU, so it's not necessarily hardware (and therefore partition) related. 

So does that mean you have to add feature hip in each environment? perhaps I did that in order to refactor the feature hip line but now that you've pointed it out, I've probably made a mistake in my thinking.

```
    'environments': [
        {
            'name': 'builtin',
            'cc': 'gcc',
            'cxx': 'g++',
            'ftn': 'gfortran',
            'target_systems': ['crusher','adastra']
        },
```

Reply:
Since in the test its use is related to compilation, I would rather use it as an environment feature.

Reply:
done